common/src/main/resources/common/common.bom

```diff
@@ -172,3 +172,49 @@ brooklyn.catalog:
           launch.command: |
             haproxy -f haproxy.conf -f servers.conf -c &&
               haproxy -D -f haproxy.conf -f servers.conf -sf $(cat ${PID_FILE})
+
+        brooklyn.initializers:
+        - type: org.apache.brooklyn.core.effector.ssh.SshCommandEffector
+          brooklyn.config:
+            name: configureHaproxy
+            description: |
+              Generate HAProxy configuration from endpoint list
+            shell.env:
+              HAPROXY_PROTOCOL: $brooklyn:config("haproxy.protocol")
+              HAPROXY_ENDPOINTS: $brooklyn:attributeWhenReady("haproxy.endpoints")
+            command: |
+              (
+              flock -o 9
+              cat > ${RUN_DIR}/servers.conf <<-EOF
+              backend servers
+              EOF
+              if [ "${HAPROXY_PROTOCOL}" == "http" ]; then
+              cat >> ${RUN_DIR}/servers.conf <<-EOF
+                mode http
+                option httpchk
+              EOF
+              fi
+              cat >> ${RUN_DIR}/servers.conf <<-EOF
+                balance roundrobin
+              EOF
+              for endpoint in $(echo ${HAPROXY_ENDPOINTS} | tr ',' ' ') ; do
+                echo "  server manager$((++n)) ${endpoint} maxconn 32 ssl \
```

Comment:
hmm right @grkvlt, I missed that in the refactoring.
How would you tackle this one? a config key to specify options such as `check` or `maxconn 32 ssl`? Or is it deducible from `HAPROXY_PROTOCOL`?
I still think it's worth generalising this entity, but only if feasible.